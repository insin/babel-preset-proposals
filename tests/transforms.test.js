import {transformSync} from '@babel/core'

import proposalsPreset from '../src'

describe('transforms', () => {
  test('class properties, decorators and export extensions', () => {
    let code = `
export thing from './module2'
export * as ns from './module1'

export @Test class TestClass {
  testProperty = true
} `
    let config = {
      presets: [
        [proposalsPreset, {
          classProperties: true,
          decorators: true,
          exportDefaultFrom: true,
          exportNamespaceFrom: true,
        }]
      ]
    }
    expect(transformSync(code, config).code).toMatchSnapshot()
  })

  test('class properties, legacy decorators and export extensions', () => {
    let code = `
export thing from './module2'
export * as ns from './module1'

@Test
export class TestClass {
  testProperty = true
} `
    let config = {
      presets: [
        [proposalsPreset, {
          classProperties: {loose: true},
          decorators: {legacy: true},
          exportDefaultFrom: true,
          exportNamespaceFrom: true,
        }]
      ]
    }
    expect(transformSync(code, config).code).toMatchSnapshot()
  })

  test('class static block and class properties', () => {
    let code = `
export class TestClass {
  static #x = 42
  static y
  static {
    this.y = this.#x * 2
  }
  testProperty = true
} `
    let config = {
      presets: [
        [proposalsPreset, {
          classStaticBlock: true,
          classProperties: true,
        }]
      ]
    }
    expect(transformSync(code, config).code).toMatchSnapshot()
  })
})
