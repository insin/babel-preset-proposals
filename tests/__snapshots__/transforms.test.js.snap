// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`transforms class properties, decorators and export extensions 1`] = `
"var _class, _temp;

import _thing from './module2';
export { _thing as thing };
import * as _ns from './module1';
export { _ns as ns };

let TestClass = Test(_class = (_temp = class TestClass {
  constructor() {
    this.testProperty = true;
  }

}, _temp)) || _class;"
`;
